### ISSUE TITLE
GenomAutomorphism

### ISSUE THREAD

**Issue by genomaths** [Bioconductor/Contributions#2678]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/genomaths/GenomAutomorphism

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by vjcitn** [comment]
Thanks for this contribution.  Your vignette has
```
URL <- paste0("https://github.com/genomaths/seqalignments/raw/master/", 
        "COVID-19/AY390556.1_and_KY417151.1_aligned_protein-coding.fas")
```
We don't allow dependence of code on data resources lodged in github.  You can contribute the data (if large) to ExperimentHub so that there is a systematic way of retrieving and caching the resource.  As it stands the
data would be read every time the vignette is built.  I also find that your vignette does not become
available when I build and install your package, via the typical method `help.start()`.  Can you check this?
It must have something to do with the VignetteEngine or builder setting....

**Comment by genomaths** [comment]
Thank you for letting me know

1. The dependence issue was fixed. The required data was already included
in the package in "/data/covid_aln.rda" (I forgot to update the vignette
with: data(covid_aln, package = "GenomAutomorphism"), now fixed.

2. In the file .Rbuildignore, I remove the line with "^vignettes/" and, in
the vignette, I introduced the change VignetteEngine{knitr::knitr} ->
VignetteEngine{knitr::rmarkdown}. Next, I just follow the suggestions given
at
https://stackoverflow.com/questions/49476812/error-in-browsevignettes-no-vignettes-found:
"*When building and reinstalling the package, it appears that sometimes
vignettes aren't included unless the install is done from a .tar.gz file"*.

a. R CMD build GenomAutomorphism.

b.  install.packages('~/GenomAutomorphism_0.99.0.tar.gz') : this creates
the folder: "/R/x86_64-pc-linux-gnu-library/4.2/GenomAutomorphism/doc"
where the vignettes are available for the user.
Next, we can just follow the steps: help.start() -> Package
-> GenomAutomorphism -> 'package vignettes and other documentation' ->
'GenomAutomorphism::GenomAutomorphism'.

Also, we can get it directly typing: vignette("GenomAutomorphism")

In summary, the installation via
install.packages('~/GenomAutomorphism_0.99.0.tar.gz') works for me.
However, the installation from inside the Rstudio project fails on building
the documentation (on Linux Ubuntu).

Hope these changes are good enough to move forward.

Best regards,
Robersy

**Comment by lshep** [comment]
sorry for the delay. I'm working on the review and should have it posted in the next day or two.

**Comment by lshep** [comment]
I'm still working on looking at the R code and class design but in the meantime here are a few easier clean up comments: 

**General**

- [ ] We encourage pkgdown configuration not to be git tracked or at least not
  included on the default branch submitted to Bioconductor. okay if you want to
  leave for now but please consider for the future.

**README**

- [ ] Areboth .md and .Rmd needed? Having both lends to one being out of date. 


- [ ] BiocManager::install will install both CRAN and Bioconductor packages so
there is no need to use install.packages.

- [ ] Please update installation to show Bioconductor installation (also for the
  future BiocManager will also install github remotes `BiocManager::install('genomaths/GenomAutomorphism')`


**Description**

- [ ] We don't recommend using `LazyData: true` unless there is a specific use
  case for it. Please remove if not needed as it can slow installation of the
  package.

**NAMESPACE**

- [ ] If you are utilizing class structures (ie. GenomeRanges, BioStrings, etc)
  than it is often advisable to import the entire package rather than selective
  import so that the full functionality of the class structure is inheritantly
  present.

**inst**

- [ ] Normally we encourage additional data or files included to be in
  subdirectories `inst/extdata` and `inst/scripts`. extdata providing the data
  files and scripts having a file that includes code, sudo-code, or text
  describing how the data was generated including relevant source information.


**vignette**

- [ ] In section 5 and 6 you are missing `package=` in the data call else there is a
  warning

```
Warning message:
In data(autm_z125, "GenomAutomorphism") :
  data set 'GenomAutomorphism' not found
```


Will have any further comments shortly.

**Comment by genomaths** [comment]
Dear Ishep
Bioconductor/Contributions

I have updated the GitHub repo introducing your suggestions.

The valid push was received on git.bioconductor.orga bout 19h ago but
apparently the building step is not moving forward.

Best regards,
Robersy

**Comment by lshep** [comment]
apologies. It looks like the builder stalled on our end.  Im rerunning the report manually and it should post shortly.  I will review the changes over the next week or so.  Cheers,

**Comment by lshep** [comment]
The master branch of your GitHub repository has been added to Bioconductor's git repository.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/genomaths.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("GenomAutomorphism")`. The package 'landing page' will be created at

https://bioconductor.org/packages/GenomAutomorphism

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
